### PR TITLE
Add run-operation to click CLI (#5552)

### LINKED ISSUE #5552: [CT-928] `dbt run-operation` works in click

### PR DESCRIPTION
**Pull request by jtcohen6**
resolves #5552

Sorry, just couldn't let others have all the fun...

### L'engagement

- Add `RunOperationTask` to click CLI
- Add `macro` positional argument

### Le tour

Borrowing 63870d5953e5c355293ae9387dfb77cc2d74b597 locally, to avoid the error `Object of type SpawnContext is not JSON serializable`:

```
$ python3 -m pytest tests/functional/run_operations/
==================================== test session starts ====================================
platform darwin -- Python 3.10.9, pytest-7.2.0, pluggy-1.0.0
rootdir: /Users/jerco/dev/product/dbt-core, configfile: pytest.ini
plugins: csv-3.0.0, logbook-1.2.0, xdist-3.1.0, flaky-3.7.0, mock-3.10.0, dotenv-0.5.2, cov-4.0.0
collected 10 items

tests/functional/run_operations/test_run_operations.py ..........                     [100%]

==================================== 10 passed in 8.47s =====================================
```

### 🎩 Le prestige

```sql
{% macro say_hello(whomst="world", strong=False) %}
    {{ print("Hello " ~ whomst ~ (" !" if strong else "")) }}
{% endmacro %}
```

```bash
$ python3 -m dbt.cli.main run-operation say_hello
Hello world
$ python3 -m dbt.cli.main run-operation project_funway.say_hello
Hello world
$ python3 -m dbt.cli.main run-operation say_hello --args 'whomst: Jeremy'
Hello Jeremy
$ python3 -m dbt.cli.main run-operation say_hello --args '{"whomst": "Wisconsin", "strong": true}'
Hello Wisconsin !
```

### Checklist

- [x] I have read [the contributing guide](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md) and understand what's expected of me
- [x] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [x] I have run this code in development and it appears to resolve the stated issue
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [x] I have [opened an issue to add/update docs](https://github.com/dbt-labs/docs.getdbt.com/issues/new/choose), or docs changes are not required/relevant for this PR
- [x] I have run `changie new` to [create a changelog entry](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md#adding-a-changelog-entry)

### CODE REVIEW

File: core/dbt/cli/main.py [hunk 1]
```diff
@@ -349,6 +350,7 @@ def run(ctx, **kwargs):
 
 # dbt run operation
 @cli.command("run-operation")
+@click.argument("macro")
```
Comment (MichelleArk) [inline core/dbt/cli/main.py:353]: tiny nit: could we move this down under `@click.pass_context` so that all the params are logically grouped together?

Comment (MichelleArk) [reply core/dbt/cli/main.py:353]: also - we should probably define this in `core/dbt/cli/params.py` alongside the other params for consistency / reuse

**Review by MichelleArk — APPROVED** [review]
just a small nit otherwise LGTM! thank you for picking this up!

File: core/dbt/cli/main.py (hunk 1 above)
Comment (jtcohen6) [reply core/dbt/cli/main.py:353]: Agree with first nit! I'll make the order match the CLI order

For the second - two reasons I prefer keeping it alongside the command definitions:
1. Click arguments can't have documentation / help text, [click's recommendation](https://click.palletsprojects.com/en/8.1.x/documentation/#documenting-arguments) is to document them by mentioning them explicitly in the command help text
2. We have very very few positional arguments in dbt commands. (I think it's just `run-operation MACRO` and `init PROJECT_NAME`. I just remembered the latter, and will add in this PR so we don't forget.) These are truly one-offs. IMO adding them as (reusable) params risks the misperception that they're widely used, versus keeping them alongside the only commands for which they're relevant.

**Pull request merged**